app/controllers/users/registrations_controller.rb

```diff
@@ -74,4 +78,37 @@ def verify_via_recaptcha!
   def handle_recaptcha_failure
     render json: {error: "There was an temporary error preventing your payment. Please try again. If it persists, please contact support@commitchange.com with error code: 5X4J "}, status: :unprocessable_entity
   end
+
+  def update_params
+    params[:user].except(:otp_required_for_login)
+  end
+
+  def handle_two_factor_changes
+    otp_param = params[:user][:otp_required_for_login]
+    return unless %w[true false].include?(otp_param)
```

Comment:
I'm not a huge fan of this, this generally should be handled by some sort of class that inherits from ActiveModel. I don't know that it's a blocker though.

Reply:
Sorry, should have been clearer: parsing the true/false value here isn't ideal.

Reply:
Now just assigning `otp_param` to `current_user.otp_required_for_login` so it'll use the same conversion logic that gets used for normal updates from params (e.g. `user.update(params)`). Then uses the `changed?` status of that field to decide whether to generate or clear the otp secret.